transformers.py

```diff
@@ -0,0 +1,155 @@
+import sonnet as snt
+import tensorflow as tf
+import math
+import numpy as np
+
+def positional_encoding(seq_len, d_model, batch_size=1):
+    """
+    Returns a tensor following the postional encoding function
+     (sinusodal from Vaswani et. all 2017).
+
+    Return shape: [batch, seq_len, d_model]
+    """
+    def encoding_angle(pos, i):
+        pos = tf.cast(pos, tf.dtypes.float32)
+        i = tf.cast(i, tf.dtypes.float32)
+        d = tf.cast(d_model, tf.dtypes.float32)
+        denom = tf.math.pow(10000., 2. * i/d)
+        return pos / denom
+
+    # BUG: Fails to serialize these calls:
+    # "Failed to convert object of type <class 'list'> to Tensor. Contents: [None, 480]. Consider casting elements to a supported type."
```

Comment:
@vladfi1 wanted to circle back on this here, I think we worked around this by disabling the compile step. Can you advise on the proper fix?